sdk/search/Azure.Search.Documents/src/AzureSearchDocumentsEventSource.cs

```diff
@@ -15,35 +18,69 @@ internal sealed class AzureSearchDocumentsEventSource : EventSource
+        public void BatchSubmitted(string publisherType, string endPoint, int batchSize)
         {
-            WriteEvent(BatchSubmittedEvent, endPoint, batchSize);
+            WriteEvent(BatchSubmittedEvent, publisherType, endPoint, batchSize);
         }
 
-        [Event(BatchActionCountUpdatedEvent, Level = EventLevel.Informational, Message = "Indexing publisher at {0} has updated the starting batch action count from {1} to {2}.")]
-        public void BatchActionCountUpdated(string endPoint, int oldBatchCount, int newBatchCount)
+        [Event(BatchActionCountUpdatedEvent, Level = EventLevel.Warning, Message = "{0} at {1} has updated the starting batch action count from {2} to {3}.")]
+        public void BatchActionCountUpdated(string publisherType, string endPoint, int oldBatchCount, int newBatchCount)
         {
-            WriteEvent(BatchActionCountUpdatedEvent, endPoint, oldBatchCount, newBatchCount);
+            WriteEvent(BatchActionCountUpdatedEvent, publisherType, endPoint, oldBatchCount, newBatchCount);
+        }
+
+        [Event(SearchIndexingBufferedSenderDisposedWithPendingActionsEvent, Level = EventLevel.Error, Message = "{0}: {2} unsent indexing actions at {1}.")]
+        public void SearchIndexingBufferedSenderDisposedWithPendingActions(string componentType, string endPoint, int indexingActionsCount)
+        {
+            WriteEvent(SearchIndexingBufferedSenderDisposedWithPendingActionsEvent, componentType, endPoint, indexingActionsCount);
+        }
+
+        [Event(DocumentsPublishedEvent, Level = EventLevel.Verbose, Message = "{0}: publishing documents. Flush = {2}")]
+        public void PublishingDocuments(string publisherType, bool flush)
+        {
+            WriteEvent(DocumentsPublishedEvent, publisherType, flush);
+        }
+
+        [Event(ActionNotificationEventHandlerExceptionThrownEvent, Level = EventLevel.Error, Message = "{0}: exception thrown for {1}. Action = {2}. Exception: {3}")]
+        public void ActionNotificationEventHandlerExceptionThrown(string componentType, string endPoint, string action, string exceptionText)
+        {
+            WriteEvent(ActionNotificationEventHandlerExceptionThrownEvent, componentType, endPoint, action, exceptionText);
+        }
+
+        [Event(BatchActionPayloadTooLargeEvent, Level = EventLevel.Warning, Message = "{0}: Batch action count {2} is too large for {1}.")]
+        public void BatchActionPayloadTooLarge(string componentType, string endPoint, int batchActionCount)
+        {
+            WriteEvent(BatchActionPayloadTooLargeEvent, componentType, endPoint, batchActionCount);
+        }
+
+        [NonEvent]
+        public void ActionNotificationEventHandlerExceptionThrown(string componentType, string endPoint, string action, Exception e)
+        {
+            ActionNotificationEventHandlerExceptionThrown(componentType, endPoint, action, e.ToString());
```

Comment:
Is this needed now?

Reply:
Methods marked with the `Event` attribute cannot take an `Exception` as a parameter.
So, either all callers can pass in messages extracted from exceptions, or we have a `NonEvent` method that does that.